### PR TITLE
Add map pool assignment

### PR DESCRIPTION
**Pull request by Sheikah45**

### CODE REVIEW

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/MapPoolAssignment.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.faforever.commons.api.dto;
+
+import com.github.jasminb.jsonapi.annotations.Id;
+import com.github.jasminb.jsonapi.annotations.Relationship;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class MapPoolAssignment {
```
Comment (Brutus5000) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/MapPoolAssignment.java]: Needs to `implements ElideEntity` and @EqualsAndHashCode for id.
Even better: let's enrich it once more so it can `extends AbstractEntity`

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/MapPoolAssignment.java (hunk 1 above)
Comment (Sheikah45) [reply faf-commons-api/src/main/java/com/faforever/commons/api/dto/MapPoolAssignment.java]: Aight

**Comment by Sheikah45** [comment]
tied to https://github.com/FAForever/faf-java-api/pull/447

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/MapParams.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.Setter;
+
+@JsonTypeInfo(use = Id.NAME, property = "type")
+@JsonSubTypes(
+    @JsonSubTypes.Type(value = NeroxisGeneratorParams.class, name = "neroxis")
+)
+@Getter
+@Setter
+@NoArgsConstructor
+public abstract class MapParams {
```
Comment (Brutus5000) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/MapParams.java]: Use an interface instead of an abstract class and remove the ununused Lombok annotations

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.faforever.commons.api.dto;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@EqualsAndHashCode(callSuper = true)
+@Getter
+@Setter
```
Comment (Brutus5000) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java]: Use a value class instead? (Maybe I'm missing a potential compile error, but I see no reason for that)

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+@JsonTypeName("neroxis")
```
Comment (Brutus5000) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java]: Why is this needed? This should be implicit by the JsonSubTypes declaration

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java (hunk 3 above)
Comment (Sheikah45) [reply faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java]: You need a no args constructor for the json deserialization I believe

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java (hunk 4 above)
Comment (Sheikah45) [reply faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java]: You are right it is not

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java (hunk 3 above)
Comment (Brutus5000) [reply faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java]: No, I think that was only the case in older Jackson versions.

Comment (Sheikah45) [reply faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java]: I tried using value yesterday and in the moderator client I got the error during runtime that it couldn't parse it due to the lack of a default constructor.

Comment (Brutus5000) [reply faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java]: As discussed adding 
lombok.anyConstructor.addConstructorProperties = true

to lombok.config will make @Value classes work with Jackson.

Comment (Sheikah45) [reply faf-commons-api/src/main/java/com/faforever/commons/api/dto/NeroxisGeneratorParams.java]: yep just forgot to push it. should be changed now

**Review by Brutus5000 — APPROVED** [review]
